wave_vision/include/wave/vision/detector/fast_detector.hpp

```diff
@@ -18,18 +18,22 @@ namespace wave {
 
 /** Configuration parameters for the FASTDetector. */
 struct FASTDetectorParams {
-    FASTDetectorParams() {}
+    FASTDetectorParams() = default;
 
-    FASTDetectorParams(int threshold, bool nonmax_suppression, int type)
+    FASTDetectorParams(int threshold,
+                       bool nonmax_suppression,
+                       int type,
+                       int num_features)
```

Comment:
const?

Reply:
But I'm changing the member variables in the initializer list. Wouldn't that not be const?

Reply:
I do not follow. You're passing by value, if we follow the "const" correct way, everything should be const if the function argument is not going to change. Which in this case threshold, nonmax_suppression, type... do not. They are passed by value to member variables.

Reply:
OH shoot yes I see, I thought you were talking about the method itself, not the arguments. thank you!

Reply:
It really isn't a big deal with these, because these are primitives, and you can only pass by value, there is no way the variables you pass in would be changed so `const` is not really necessary.

But! IMO, if we are to follow `const` correct-ness we should put `const` infront of primitives anyways, just so that you don't have to think much.

I find this SO answer provides a good rationale why even primitives should have `const`.
https://stackoverflow.com/questions/16033751/c-const-modifier-with-primitive-types